uaclient/clouds/identity.py

```diff
@@ -58,12 +60,20 @@ def get_cloud_type_from_result_file(
     return DATASOURCE_TO_CLOUD_ID.get(dsname, dsname)
 
 
+@lru_cache(maxsize=None)
 @apply_config_settings_override("cloud_type")
 def get_cloud_type() -> "Optional[str]":
     if util.which("cloud-id"):
         # Present in cloud-init on >= Xenial
-        out, _err = util.subp(["cloud-id"])
-        return out.strip()
+        try:
+            out, _err = util.subp(["cloud-id"])
+            return out.strip()
+        except util.ProcessExecutionError:
+            logging.warning(
+                "Warning: could not determine cloud ID. "
+                "If you didn't disable cloud-init, "
+                "check cloud-init logs for more information."
+            )
```

Comment:
I think we should return something here, right ? Otherwise we will run the rest of the code unnecessarily

Reply:
Also, I think we should have an integration test that explores that situation. We can manually delete the `instance-data.json` file and verify how the warning will be presented to the user

Reply:
I would also tweak the message slightly. Maybe something like:
```
Warning: cloud-init is present but did not run. Assuming not on cloud. If you are on a cloud, you may experience unreliable behavior.
```

Open to additionaly feedback/bikeshedding

Reply:
Thank you for the initial review, all three comments make all the sense.
/ me on that